### PR TITLE
Finish app (maybe?)

### LINKED ISSUE #5: Make reusable input component

### PR DESCRIPTION
**Pull request by devinbroussard**
Added styling for the ingredients input, find recipe button, and the recipe display. Also added UI to display the loading and error states.

After this PR I think the project will be in a state where we can call it finished if we no longer want to work on it.

Recipe display:
image

Loading display:
image

Error display:
image

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,38 +1,24 @@
...
+:root {
+  --font-family: 'Inter', sans-serif;
 }
 
-@media (prefers-reduced-motion: no-preference) {
-  .App-logo {
-    animation: App-logo-spin infinite 20s linear;
-  }
+html {
+  height: 100%;
```
Comment (taylorwaddell) [inline src/App.css:8]: Why?

File: src/App.css [hunk 2]
```diff
@@ -1,38 +1,24 @@
...
+  margin: 0;
+  font-size: 1.125rem;
+  font-family: var(--font-family);
 }
 
-.App-link {
-  color: #61dafb;
+input {
+  font-family: var(--font-family);
```
Comment (taylorwaddell) [inline src/App.css]: You set this in the body, I don't think it also needs to be set here..?

File: src/App.css [hunk 3]
```diff
@@ -1,38 +1,24 @@
...
 }
 
-@keyframes App-logo-spin {
-  from {
-    transform: rotate(0deg);
-  }
-  to {
-    transform: rotate(360deg);
-  }
+#root {
```
Comment (taylorwaddell) [inline src/App.css:22]: Targeting all elements with an ID of "root"? Was that intentional. I believe you meant to apply this to the `:root`. But you shouldn't have to apply `height: 100%;` to the root. Do you need this?

File: src/api/OpenAIAPI.js [hunk 4]
```diff
@@ -6,24 +6,23 @@ const openAI = new OpenAI({
...
+    throw new Error('There was an generating error the recipe. Please try again.');
 
-    const recipeObject = JSON.parse(response.choices[0].message.content);
-    return recipeObject;
-  } catch (error) {
-    return { error: `There was an error generating the recipe. Please try again. ${error}`};
-  }
+  const recipeObject = JSON.parse(response.choices[0].message.content);
+  if (!recipeObject.name || !recipeObject.ingredients || !recipeObject.instructions || !recipeObject.cookTime)
+    throw new Error('There was an error generating the recipe. Please try again.');
```
Comment (taylorwaddell) [inline src/api/OpenAIAPI.js]: You use this same string twice. Make variable?

File: src/api/OpenAIAPI.js [hunk 5]
```diff
@@ -6,24 +6,23 @@ const openAI = new OpenAI({
...
+
+  return recipeObject;
 }
 
 function getContent(ingredients) {
-  return `I'm building a recipe finder app where users input ingredients, and I provide them with a recipe they can make using those ingredients. Please provide a recipe in JSON format with the following keys: name (string), ingredients (string[]), instructions (string[]), and cookTime (string). Ingredients: ${ingredients}`;
+  return `I'm building a recipe finder app where users input ingredients, and I provide them with a recipe they can make using those ingredients. Please provide a recipe in JSON format with the following keys: name (string), ingredients (string[]), instructions (string[]), and cookTime (string). Do not put numbers in the instructions. Ingredients: ${ingredients}`;
```
Comment (taylorwaddell) [inline src/api/OpenAIAPI.js]: Have you tried asking ChatGPT to improve this prompt? You may be able to make this prompt a little stronger.

File: src/pages/RecipeFinder.js [hunk 6]
```diff
@@ -1,44 +1,86 @@
...
+        <span className='input-with-btn'>
+          <input
+            type="text"
+            placeholder="Enter ingredients"
+            value={ingredients}
+            onChange={handleInputChange}
+            className='dish-input'
+            required
+          />
+          <button className='find-recipe-btn' disabled={!ingredients} type="submit">
```
Comment (taylorwaddell) [inline src/pages/RecipeFinder.js]: This would still be enabled if they typed in an empty space

File: src/styles/RecipeFinder.css [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  padding: 0.2rem;
+  border: 0;
+  font-size: 1.3rem;
+  &:focus {
+    border: 0
+  }
+}
+
+.dish-input:focus-visible {
+  outline: none;
```
Comment (taylorwaddell) [inline src/styles/RecipeFinder.css:11]: Why remove the focus outline?

File: src/styles/RecipeFinder.css [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+.find-recipe-btn {
+  font-size: 1.2rem;
+  border-radius: 0.75rem;
+  cursor: pointer;
+  background-color: rgb(86, 155, 59);
+  color: white;
+  border: 0.1rem solid rgb(214, 209, 209);
+  padding: 0.4rem;
```
Comment (taylorwaddell) [inline src/styles/RecipeFinder.css]: You could simplify your padding changes as such: `padding: 0.4rem 0.7rem`

File: src/styles/RecipeFinder.css [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  padding-left: 0.7rem;
+  padding-right: 0.7rem;
+
+  font-family: "Inter", sans-serif;
```
Comment (taylorwaddell) [inline src/styles/RecipeFinder.css]: This was already set in the `body`. Do you need this for the inner input text? Is that why you did it?

File: src/styles/RecipeFinder.css [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+.find-recipe-btn:disabled {
```
Comment (taylorwaddell) [inline src/styles/RecipeFinder.css:28]: This just came to mind so you'd have to Google it. I wonder if it's good practice to put `pointer-events: none;` on disabled buttons...?

File: src/styles/RecipeFinder.css [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+.input-with-btn {
+  border: 0.1rem solid rgb(214, 209, 209);
+  padding: 0.4rem;
+  border-radius: 0.75rem;
+  display: flex;
+}
+
+.container {
+  min-height: 80%;
+  padding-top: 6rem;
```
Comment (taylorwaddell) [inline src/styles/RecipeFinder.css]: You can simplify your padding as such: `padding: 6rem 1rem 1rem 0rem;`

File: src/styles/RecipeFinder.css [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  padding-left: 1rem;
+  padding-right: 1rem;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+}
+
+.form {
```
Comment (taylorwaddell) [inline src/styles/RecipeFinder.css]: You use this exact same set of rules on the above `.container` class as well. Maybe just generalize this class name and use it for both things?

**Review by taylorwaddell — COMMENTED** [review]
I don't think I see anything that would break. Just a couple thoughts, questions, and suggestions (:

File: src/App.css (hunk 1 above)
Comment (devinbroussard) [reply src/App.css:8]: Did this so that I could center things vertically on the page

File: src/App.css (hunk 3 above)
Comment (devinbroussard) [reply src/App.css:22]: This was to apply 100% height to React's wrapper element that has an id of root. I used it to be able to center things vertically, like with the 100% on the html/body elements

File: src/App.css (hunk 2 above)
Comment (devinbroussard) [reply src/App.css]: Apparently `input` and `button` elements don't inherit the font from the `body` element by default, so I set `font-family: inherit` for both of these

File: src/styles/RecipeFinder.css (hunk 7 above)
Comment (devinbroussard) [reply src/styles/RecipeFinder.css:11]: Added outline on the input's container instead

File: src/styles/RecipeFinder.css (hunk 10 above)
Comment (devinbroussard) [reply src/styles/RecipeFinder.css:28]: Added

File: src/styles/RecipeFinder.css [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+.container {
+  min-height: 80%;
+  padding: 6rem 1rem 1rem 0rem;
+}
+
+.recipe-details {
+  max-width: 40rem;
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 3rem;
```
Comment (taylorwaddell) [inline src/styles/RecipeFinder.css]: not necessary but you could do `margin: 3rem auto;`

**Review by taylorwaddell — APPROVED** [review]
Added one comment on a margin that could be consolidated if you wanted. Otherwise, lgtm!

**Pull request merged**